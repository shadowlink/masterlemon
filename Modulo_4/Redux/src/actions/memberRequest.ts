import { actionsEnums } from "../common/actionsEnums";
import { MemberEntity } from "../model/member";
import { memberAPI } from "../api/member";

export const memberRequestCompleted = (members: MemberEntity[]) => {
  return {
    type: actionsEnums.MEMBER_REQUEST_COMPLETED,
    payload: members
  };
};

export const memberRequest = (organizationName: string) => dispatcher => {
  const promise = memberAPI.getAllMembers(organizationName);

  promise.then(data => dispatcher(memberRequestCompleted(data)));

  return promise;
};

export const updateOrganizationField = (organizationName: string) => {
  return {
    type: actionsEnums.UPDATE_ORGANIZATION_FIELD,
    payload: organizationName
  };
};
