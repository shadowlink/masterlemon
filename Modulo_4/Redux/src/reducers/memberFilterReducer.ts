import { actionsEnums } from "../common/actionsEnums";
import { FilterEntity } from "../model/filter";

export type FilterState = FilterEntity;

export const memberFilterReducer = (
  state: FilterState = { organizationField: "lemoncode" },
  action
) => {
  switch (action.type) {
    case actionsEnums.UPDATE_ORGANIZATION_FIELD:
      return handleUpdateOrganizationFilterState(state, action.payload);
  }

  return state;
};

const handleUpdateOrganizationFilterState = (
  state: FilterState,
  organizationName: string
) => ({
  ...state,
  organizationField: organizationName
});
