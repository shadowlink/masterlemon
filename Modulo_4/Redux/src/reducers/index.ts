import { combineReducers } from "redux";
import { memberReducer, memberState } from "./memberReducer";
import { FilterState, memberFilterReducer } from "./memberFilterReducer";

export interface State {
  memberReducer: memberState;
  memberFilterReducer: FilterState;
}

export const reducers = combineReducers<State>({
  memberReducer,
  memberFilterReducer
});
