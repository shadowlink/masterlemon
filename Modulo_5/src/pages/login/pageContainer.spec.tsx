import * as React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { State } from '../reducers';
import { createEmptyLoginFormErrors, createEmptyLoginEntity } from './viewModel';
import * as updateFieldActions from './actions/updateLoginEntityField';
import * as loginRequestActions from './actions/loginRequest';
import { LoginPageContainer } from './pageContainer';

const getMockStore = configureStore();

describe('pages/login/pageContainer tests', () => {
  it('should render as expected passing state', () => {
    // Arrange
    const state = {
      login: {
        loginEntity: createEmptyLoginEntity(),
        loginFormErrors: createEmptyLoginFormErrors(),
      },
    } as State;

    const store = getMockStore(state);

    // Act
    const component = shallow(
      <LoginPageContainer
      />,
      {
        context: { store },
      }
    );

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('should call to updateLoginEntityField action creator when call to updateField prop', () => {
    // Arrange
    const state = {
      login: {
        loginEntity: createEmptyLoginEntity(),
        loginFormErrors: createEmptyLoginFormErrors(),
      },
    } as State;

    const store = getMockStore(state);
    const actionCreatorStub = jest.spyOn(updateFieldActions, 'updateLoginEntityField')
      .mockImplementation(() => ({
        type: 'test action type',
      }));

    // Act
    const component = shallow(
      <LoginPageContainer
      />,
      {
        context: { store },
      }
    );

    component.prop('updateField')('test fieldName', 'test value');

    // Assert
    expect(actionCreatorStub).toHaveBeenCalledWith(state.login.loginEntity, 'test fieldName', 'test value');
  });

  it('should call to loginRequest action creator when call to doLogin prop', () => {
    // Arrange
    const state = {
      login: {
        loginEntity: createEmptyLoginEntity(),
        loginFormErrors: createEmptyLoginFormErrors(),
      },
    } as State;

    const store = getMockStore(state);
    const actionCreatorStub = jest.spyOn(loginRequestActions, 'loginRequest')
      .mockImplementation(() => ({
        type: 'test action type',
      }));

    // Act
    const component = shallow(
      <LoginPageContainer
      />,
      {
        context: { store },
      }
    );

    component.prop('doLogin')();

    // Assert
    expect(actionCreatorStub).toHaveBeenCalledWith(state.login.loginEntity);
  });
});
