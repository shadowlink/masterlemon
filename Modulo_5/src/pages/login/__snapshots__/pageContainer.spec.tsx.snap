// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/login/pageContainer tests should render as expected passing state 1`] = `
<Component
  doLogin={[Function]}
  loginEntity={
    Object {
      "login": "",
      "password": "",
    }
  }
  loginFormErrors={
    Object {
      "login": FieldValidationResult {
        "errorMessage": "",
        "key": "",
        "succeeded": false,
        "type": "",
      },
      "password": FieldValidationResult {
        "errorMessage": "",
        "key": "",
        "succeeded": false,
        "type": "",
      },
    }
  }
  updateField={[Function]}
/>
`;
